ee/localserver/zta.go

```diff
@@ -3,12 +3,31 @@ package localserver
 import (
 	"log/slog"
 	"net/http"
+	"strings"
 
 	"github.com/kolide/launcher/pkg/traces"
 )
 
 var (
 	localserverZtaInfoKey = []byte("localserver_zta_info")
+
+	// allowlistedZtaOriginsLookup contains the complete list of origins that are permitted to access the /zta endpoint.
+	allowlistedZtaOriginsLookup = map[string]struct{}{
+		// Release extension
+		"chrome-extension://gejiddohjgogedgjnonbofjigllpkmbf":         {},
+		"chrome-extension://khgocmkkpikpnmmkgmdnfckapcdkgfaf":         {},
+		"chrome-extension://aeblfdkhhhdcdjpifhhbdiojplfjncoa":         {},
+		"chrome-extension://dppgmdbiimibapkepcbdbmkaabgiofem":         {},
+		"moz-extension://dfbae458-fb6f-4614-856e-094108a80852":        {},
+		"moz-extension://25fc87fa-4d31-4fee-b5c1-c32a7844c063":        {},
+		"moz-extension://d634138d-c276-4fc8-924b-40a0ea21d284":        {},
+		"safari-web-extension://7cba6ce0-bc55-43bd-b2a3-562b703ecbea": {},
```

Comment:
As noted over Slack, we'll need to prefix match `safari-web-extension://` instead of hardcode an expected value since the UUID for an SWE is randomized each time that SWE is loaded.

Reply:
Addressed: https://github.com/kolide/launcher/pull/2117/commits/7eb33b1903ff9696f786965dd2129b627e0755a3

Reply:
> prefix match safari-web-extension:// 

Doesn't this allow _all_ safari extensions, thus undermining the point?

Reply:
This is only going to be one defense in depth layer but more importantly: Every SWE is a native code bundle by definition (they are a special app bundle installed via the app store), so just as `Origin` matching isn't a strong defense against Chrome/Firefox extensions with NMH or native apps, it isn't a strong defense against Safari extensions. These Safari extensions can run HTTP requests from their Swift code and spoof their `Origin` if desired.

Reply:
@directionless -- yes, unfortunately it will allow all safari extensions. We're still preventing any other random site from accessing `/zta`, so this is an improvement, albeit incomplete. I think we've got to call this "good enough for now".

Reply:
If @complexspaces is good, I'm good. I just want to note it.